pkg-manager/core/src/install/index.ts

```diff
@@ -566,14 +572,31 @@ export async function mutateModules (
         optionalDependencies,
         updateWorkspaceDependencies: project.update,
         preferredSpecs,
-        saveCatalogName: opts.saveCatalogName,
+        saveCatalogName,
         overrides: opts.overrides,
         defaultCatalog: opts.catalogs?.default,
       })
+
       projectsToInstall.push({
         pruneDirectDependencies: false,
         ...project,
-        wantedDependencies: wantedDeps.map(wantedDep => ({ ...wantedDep, isNew: !currentBareSpecifiers[wantedDep.alias], updateSpec: true, nodeExecPath: opts.nodeExecPath })),
+        wantedDependencies: wantedDeps.map(wantedDep => {
+          if (saveCatalogName && !wantedDep.bareSpecifier.includes(':')) {
+            const catalogDep = opts.catalogs?.[saveCatalogName]?.[wantedDep.alias]
+            if (catalogDep && !intersects(wantedDep.bareSpecifier, catalogDep)) {
+              if (opts.catalogMode === 'strict') {
```

Comment:
I'd recommend an exhaustive `switch` here to force a compile error. It's more defensive in case new catalog modes are added in the future.

```js
switch (opts.catalogMode) {
  case 'strict':
    throw new CatalogVersionMismatchError({ catalogDep: `${wantedDep.alias}@${catalogDep}`, wantedDep: `${wantedDep.alias}@${wantedDep.bareSpecifier}` })

  case 'recommended':
  case 'manual'
    // ...
}
```

If a new mode is added, we want to force the developer to consider what changes they should make to this block.